test/ui-test/Jenkinsfile

```diff
@@ -0,0 +1,357 @@
+            result=1
+        else
+            result=0
+            break;
+        fi
+    done
+    echo \$result
+}
+
+
+
+cd \${WORKSPACE2}/crc
+
+## Start crc and install all necessary stuff
+
+./\${BASEFILE_NAME} start -p \${CRC_PULL_SECRET} --memory 20480 --cpus 10
+
+./\${BASEFILE_NAME} status || true
+
+# verify crc is started
+echo "CRC is starting..."
+treshold=180
+timer=0
+starting_result="CRC is started and ready"
+while [ "\$(crc_started)" != "1" ]; do
+	echo "waiting for \$timer sec..."
+    sleep 10
+    ((timer=timer+10))
+    if [ \$timer -ge \$treshold ]; then
+        starting_result="Timeout reached when starting CRC"
+        echo \$starting_result
+        exit 1
+    fi
+done
+
+echo \$starting_result
+
+# add oc to the path
+./\${BASEFILE_NAME} oc-env
+eval \$(./\${BASEFILE_NAME} oc-env)
+
+oc version
+
+# log into a cluster using oc
+# password can be obtained from ~/.crc/cache/crc_libvirt_4.5.1/kubeadmin-password
+# 4.5.1 string can be get from ./crc version or from client version: oc version
+pass=`cat ~/.crc/cache/crc_libvirt_\$(oc version | head -n 1 | cut -d':' -f2 | xargs)/kubeadmin-password`
+oc login -u kubeadmin -p \${pass} https://api.crc.testing:6443 --insecure-skip-tls-verify
+
+# to list available operators
+oc get packagemanifests -n openshift-marketplace
+
+# we are searching for serverless operator: serverless-operator
+oc describe packagemanifests serverless-operator -n openshift-marketplace
+
+# Switch to openshif
+# Create a subscription yaml object - look for spec part in installed opeator to find installCSV, etc...
+cat >> subscription-serverless.yaml <<EOL
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: serverless-operator
+  namespace: openshift-operators 
+spec:
+  channel: '4.5'
+  installPlanApproval: Automatic
+  name: serverless-operator
+  source: redhat-operators
+  sourceNamespace: openshift-marketplace
+  startingCSV: serverless-operator.v1.7.2
+EOL
+
+# apply subscription
+oc apply -f subscription-serverless.yaml
+
+# change to default project - openshift-operators
+oc project openshift-operators
+# Verification of installed operators:
+oc get subscriptions
+# serverless-operator   serverless-operator   redhat-operators   4.5
+oc get csv
+
+echo "Operator deployed function: \$(operator_deployed)"
+
+while [ "\$(operator_deployed)" != "1" ]; do
+	echo "waiting 10 sec..."
+    sleep 10
+done
+
+echo "Operator is deployed"
+
+# Create a namespace and install knative serving
+oc create namespace knative-serving
+
+# Install Knative Serving
+cat > serving.yaml << EOL
+apiVersion: operator.knative.dev/v1alpha1
+kind: KnativeServing
+metadata:
+ name: knative-serving
+ namespace: knative-serving
+EOL
+
+# apply
+oc apply -f serving.yaml
+
+# Verification
+oc get knativeserving/knative-serving -n knative-serving --template='{{range .status.conditions}}{{printf "%s=%s\\n" .type .status}}{{end}}'
+oc get pods -n knative-serving
+
+sleep 60
```

Comment:
It is default wait/sleep so it got some time at the start. Let's say it is do-while.
But it can be updated, to just keep a while check there. Same for eventing.